app/docs/src/reference/authentication.md

```diff
@@ -0,0 +1,107 @@
+# Authentication Functions
+
+Authentication functions are a kind of [function](./function.md) which handle
+credential validation and setup before other functions execute. They run
+automatically when a function needs to use [secrets](./secrets.md), ensuring
+that authentication happens transparently and consistently.
```

Comment:
I like the use of "transparent" to mean that we should how authentication works and it isn't a black box. I do think we should be clear that their secrets themselves aren't transparent 🤣

Maybe we should say "transparently, securely and consistently"?